ppmpy/ppm.py

```diff
@@ -12148,6 +12151,80 @@ def sphericalHarmonics_lmax(self, radius):
 
         return lmax, N, npoints
 
+    def get_power_spectrum(self, varloc, dump_start, dump_stop, dump_step=1, radius=None, mass=None):
+        '''
+        Calculates the power spectrum of variable 'varloc' at radius 'radius' for
+        dumps arange(dump_start, dump_stop+1, dump_step).
+        The power spectrum is expressed as power per spherical harmonic mode 'ell'.
+        
+        Note that power spectrums of velocities will be in 10^-12 m^2/s^2/ell
+
+        Parameters
+        ----------
+        varloc: str, int
+            String: for the variable you want if defined on instantiation
+            Int: index location of the variable you want the power spectrum of
+        dump_start: int
+            Dump number of the first dump to consider
+        dump_stop: int
+            Dump number of the last dump to consider
+        dump_step: int
+            Number of dumps between each dump to consider in the dump averaging
+        radius: float, optiinal
+            The radius of the sphere at which you want the power spectrum of 'varloc', in Mm
+            Note that if no radius is specified, then a mass must be specified
+        mass: float, optional
+            Mass M(R) at which to calculate the spectrum, in Msun
+            Note that if no mass is specified, then a radius must be specified
+
+        Returns
+        -------
+        ell: np.ndarray
+            Spherical harmonics modes ell
+        power_ell: np.ndarray
+            Power per model ell
+        '''
+
+        if (radius==None and mass==None) or (radius!=None and mass!=None):
+            raise ValueError('You must select wither a radius or a mass where to calculate the spectrum')
+        
+        dumps = np.arange(dump_start, dump_stop+1, dump_step)
+        
+        radiusinput = radius
+        massinput = mass
+
+        for i,dump in enumerate(tqdm(dumps)):
```

Comment:
we have runs where some moms or rprofs are missing; if you can make it so that a missing dump will be just ignored that would save us grief in the future, use
```Python
try:
   ....
except ....Error:
   ...
```

Reply:
Done.